ui/src/app/subnets/views/SubnetDetails/SubnetDetailsHeader/SubnetActionForms/components/EditBootArchitectures/BootArchitecturesTable/BootArchitecturesTable.tsx

```diff
@@ -0,0 +1,130 @@
+import { MainTable } from "@canonical/react-components";
+import classNames from "classnames";
+import { useFormikContext } from "formik";
+import { useSelector } from "react-redux";
+
+import type { FormValues } from "../EditBootArchitectures";
+
+import RowCheckbox from "app/base/components/RowCheckbox";
+import { knownBootArchitectures as knownBootArchitecturesSelectors } from "app/store/general/selectors";
+import type { KnownBootArchitecture } from "app/store/general/types";
+
+export enum Headers {
+  BootloaderArch = "Bootloader architecture",
+  BootMethod = "BIOS boot method",
+  Name = "Name",
+  Octet = "Architecture octet",
+  Protocol = "Protocol",
+}
+
+export const BootArchitecturesTable = (): JSX.Element => {
+  const {
+    setFieldValue,
+    values: { disabled_boot_architectures },
+  } = useFormikContext<FormValues>();
+  const knownBootArchitectures = useSelector(
+    knownBootArchitecturesSelectors.get
+  );
+
+  const isChecked = (bootArchName: KnownBootArchitecture["name"]) =>
+    !disabled_boot_architectures.includes(bootArchName);
+
+  const handleArchChange = (bootArchName: KnownBootArchitecture["name"]) => {
+    setFieldValue(
```

Comment:
It's probably also worth adding tests for checking and unchecking items because there is a bit of logic in this handler.